### PR TITLE
feat(linter): Implement plugin-jsdoc/check-property-names

### PR DESCRIPTION
**Pull request by leaysgur**
Part of #1170 

- Doc: https://github.com/gajus/eslint-plugin-jsdoc/blob/main/docs/rules/check-property-names.md#repos-sticky-header
- Test: https://github.com/gajus/eslint-plugin-jsdoc/blob/main/test/rules/assertions/checkPropertyNames.js
- Impl: https://github.com/gajus/eslint-plugin-jsdoc/blob/main/src/rules/checkPropertyNames.js

### CODE REVIEW

File: crates/oxc_linter/src/snapshots/check_property_names.snap [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+ 6 │                        * @property {string} employees[].department - The employee's department.
+   ·                                             ──────────────────────
+ 7 │                        */
+   ╰────
+  help: @property path declaration `employees[].department` appears before any real property.
+
+  ⚠ eslint-plugin-jsdoc(check-property-names): Duplicate @property found.
+   ╭─[check_property_names.tsx:5:27]
+ 4 │                        * @property foo
+ 5 │                        * @property foo
```
Comment (Boshen) [inline crates/oxc_linter/src/snapshots/check_property_names.snap:44]: There's a trick for labelling multiple spans https://github.com/oxc-project/oxc/blob/6532799ff29bf39a562c0d09a83cea9c7f1ee142/crates/oxc_linter/src/rules/eslint/no_sparse_arrays.rs#L58-L63

Comment (leaysgur) [reply crates/oxc_linter/src/snapshots/check_property_names.snap:47]: Wow, I didn't know that! 😮 => https://github.com/oxc-project/oxc/pull/2989/commits/d112048d599a8cf39b3fe244930456cc33e65ff3

File: crates/oxc_linter/src/rules/jsdoc/check_property_names.rs [hunk 2]
```diff
@@ -80,13 +77,22 @@ impl Rule for CheckPropertyNames {
...
+                        severity = Severity::Warning,
+                        labels = [duplicated_span, &name_part.span]
+                            .iter()
+                            .map(|span| LabeledSpan::at(
+                                (span.start as usize)..(span.end as usize),
+                                "Duplicated property".to_string(),
+                            ))
+                            .collect::<Vec<_>>(),
+                        help = format!("@property `{type_name}` is duplicated on the same block."),
+                        "eslint-plugin-jsdoc(check-property-names): Duplicate @property found."
```
Comment (Boshen) [inline crates/oxc_linter/src/rules/jsdoc/check_property_names.rs:91]: Oh, I thought there are more than 2.

For only 2 spans, e.g.:

```
    #[derive(Debug, Error, Diagnostic)]
    #[error("Duplicated export '{0}'")]
    #[diagnostic()]
    struct DuplicateExport(
        CompactStr,
        #[label("Export has already been declared here")] Span,
        #[label("It cannot be redeclared here")] Span,
    );
```

Comment (leaysgur) [reply crates/oxc_linter/src/rules/jsdoc/check_property_names.rs]: Thanks for the suggestions!

> collect all duplicates and emit them all at once at the end.

I'll go this way. 👉🏻  => https://github.com/oxc-project/oxc/pull/2989/commits/e253a071515b1044cf7b14023f47b349cd969732

**Review by Boshen — APPROVED** [review]
Thank you!

**Pull request merged**